storage/innobase/os/os0file.cc

```diff
@@ -152,6 +157,22 @@ class io_slots
 static io_slots *read_slots;
 static io_slots *write_slots;
 
+/**
+  Statistics for asynchronous I/O
+  @param[in] op operation type (aio_opcode::AIO_PREAD or aio_opcode::AIO_PWRITE)
+  @param[in] stats pointer to the structure to fill
+*/
+void innodb_io_slots_stats(tpool::aio_opcode op, innodb_async_io_stats_t *stats)
+{
+   io_slots *slots= op == tpool::aio_opcode::AIO_PREAD? read_slots : write_slots;
+
+   stats->pending_ops = slots->pending_io_count();
+   stats->slot_wait_time_sec=
+       std::chrono::duration_cast<std::chrono::duration<float>>(
+           slots->wait_time()).count();
```

Comment:
`stats->slot_wait_time_sec` is `double`, but the `slots->wait_time()` is only `float`.

Reply:
Yes? I think the underlying type is float, but casting float to double should be ok. There are only double status variables, no floats.

Reply:
Could we collect it in `double` precision?